eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/FlatCollectIterable.java

```diff
@@ -78,6 +79,24 @@ public V detect(Predicate<? super V> predicate)
     }
 
     @Override
+    public Optional<V> detectOptional(Predicate<? super V> predicate)
+    {
+        V[] result = (V[]) new Object[1];
+        Iterate.detectOptional(this.adapted, each -> {
```

Comment:
Actually you are right. Here are how I fixed this.

1. I added missing test case in FlatCollectIterableTestNoIteratorTest.
2. I applied your suggestion to use anySatisfy() for both detect() and detectOptional()
3. Fixed the logic to explicitly throw NPE when iterable is null (otherwise it'll throw IllegalArgumentException)
4. Fixed the logic to explicitly throw NPE when detecting null value